flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunner.java

```diff
@@ -112,6 +114,13 @@ public TaskManagerRunner(
 		// Initialize the TM metrics
 		TaskExecutorMetricsInitializer.instantiateStatusMetrics(taskManagerMetricGroup, taskManagerServices.getNetworkEnvironment());
 
+		HeartbeatManagerImpl heartbeatManager = new HeartbeatManagerImpl(
+				taskManagerConfiguration.getTimeout().toMilliseconds(),
+				resourceID,
+				executor,
+				Executors.newSingleThreadScheduledExecutor(),
```

Comment:
Maybe we could add to the `RpcService` that it provides a `ScheduledExecutorService`. If we see that the load on this executor is too much, then we can introduce a dedicated service. What do you think?